src/modules/hip/kernel/fog.hpp

```diff
@@ -0,0 +1,323 @@
+#include <random>
+#include <hip/hip_runtime.h>
+
+#include "rpp_hip_common.hpp"
+#include "fog_mask.hpp"
+
+__device__ __forceinline__ void fog_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *maskAlpha_f8, d_float8 *maskIntensity_f8, float4 *gamma_f4)
+{
+    float4 alphaFactor_f4[2];
+    alphaFactor_f4[0] = rpp_hip_pixel_check_0to1(maskAlpha_f8->f4[0] + *gamma_f4);
+    alphaFactor_f4[1] = rpp_hip_pixel_check_0to1(maskAlpha_f8->f4[1] + *gamma_f4);
+    dst_f8->f4[0] = (src_f8->f4[0] * ((float4)1 - alphaFactor_f4[0])) + (maskIntensity_f8->f4[0] * alphaFactor_f4[0]);
```

Comment:
we need to have a check of 0to255 for U8 for the outputs
rpp_hip_pixel_check_0to255